src/blog/2020/001-dmn-js-8.md

```diff
@@ -0,0 +1,103 @@
+---
+
+title: DMN 1.3 Support in dmn-js
+layout: blogpost.hbs
+
+slug: 2020-dmn-js-8-0-0
+author: Philipp Fromme<https://github.com/philippfromme>
+
+published: 2020-02-28 12:00
+
+releases:
+  - 'dmn-js@8.0.0'
+
+---
+
+
+<p class="introduction">
+  We're excited to announce the latest release of our [DMN modeling toolkit](https://github.com/bpmn-io/dmn-js). The release supports DMN 1.3, the next version of the DMN standard. Facilitating the latest DMN features, we now export graphical information in a standardized format. Along with this release, we publish a migration path that allows you to open DMN 1.1 diagrams using the toolkit or batch migrate all your DMN 1.1 diagrams to DMN 1.3.
+</p>
+
+<!-- continue -->
+
+Starting with DMN 1.2, the standard defines diagram interchange (DMN DI), a standardized way to specify graphical information of diagrams. In DMN 1.1 that capability was missing, so older versions of our toolkit represented graphical information via a vendor extension.
+
+Starting with `dmn-js@8` we read and write DMN 1.3 files and use the DMN standard facilities access graphical information:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<definitions
+  xmlns="https://www.omg.org/spec/DMN/20191111/MODEL/"
+  xmlns:dmndi="https://www.omg.org/spec/DMN/20191111/DMNDI/"
+  xmlns:dc="http://www.omg.org/spec/DMN/20180521/DC/"
+  id="Definitions_1"
+  name="Definitions_1">
+  <decision id="Decision_1" name="Decision_1" />
+  <dmndi:DMNDI>
+    <dmndi:DMNDiagram id="DMNDiagram_0docyxs">
+      <dmndi:DMNShape id="DMNShape_1gx1owo" dmnElementRef="Decision_1">
+        <dc:Bounds height="80" width="180" x="100" y="100" />
+      </dmndi:DMNShape>
+    </dmndi:DMNDiagram>
+  </dmndi:DMNDI>
+</definitions>
+```
+
+Checkout [this diff](https://gist.github.com/nikku/1b5bbb3ea929d2f64e5338fe6fe8b57b/revisions
+) for a before-after comparison. Try out modeling DMN 1.3 [on our demo](https://demo.bpmn.io/dmn/s/check-order).
+
+
+## Opening DMN 1.1 and DMN 1.2 Files
+
+As a significant, breaking change, __the library reads and writes DMN 1.3 files only__.
+
+We provide a migration utility, [`@bpmn-io/dmn-migrate`](https://github.com/bpmn-io/dmn-migrate) that transforms a valid DMN 1.1 diagram, including our custom DI extension into valid and standards-compliant DMN 1.3.
+
+To open older DMN files (DMN 1.1, DMN 1.2), use the utility as a pre-processor:
+
+```javascript
+import { migrateDiagram } from '@bpmn-io/dmn-migrate';
+
+// migrate diagram to DMN 1.3, if it is an older DMN version file
+const dmn1_3_XML = await migrateDiagram(dmnXML);

```

Comment:
Any specific reason for using snake_case?

Reply:
No particular reason :eyes:

Reply:
🐍